src/graph/optimizer/rule/OptimizeEdgeIndexScanByFilterRule.cpp

```diff
@@ -118,7 +118,7 @@ StatusOr<TransformResult> OptimizeEdgeIndexScanByFilterRule::transform(
         if (static_cast<ListExpression*>(inExpr->right())->size() > 1) {
           return TransformResult::noTransform();
         } else {
-          transformedExpr = graph::ExpressionUtils::rewriteInExpr(condition);
+          transformedExpr = graph::ExpressionUtils::rewriteInExpr(inExpr);
```

Comment:
This branch won't be reached for the case you added below because there is only 1 element in the list of the IN expr